nemo/collections/asr/parts/mixins/streaming.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+
+import torch
+
+
+class StreamingEncoder(ABC):
+    @abstractmethod
+    def setup_streaming_params(
+        self, init_chunk_size=None, init_shift_size=None, chunk_size=None, shift_size=None, cache_drop_size=None
+    ):
```

Comment:
All classes should have proper docstrings. Mixin classes top level docstring should document 

1) Any parameters they set to self
2) What the methods do (in the method docstrings)

For example, look at AdapterModuleMixin or ASRBPEMixin.